modules/collector-source/pkg/metric/repository.go

```diff
@@ -2,40 +2,122 @@ package metric
 
 import (
 	"fmt"
+	"path"
+	"sort"
+	"strings"
 	"sync"
 	"time"
 
+	"github.com/opencost/opencost/core/pkg/exporter"
+	"github.com/opencost/opencost/core/pkg/exporter/pathing"
 	"github.com/opencost/opencost/core/pkg/log"
+	"github.com/opencost/opencost/core/pkg/storage"
+	"github.com/opencost/opencost/core/pkg/util/json"
 	"github.com/opencost/opencost/modules/collector-source/pkg/util"
 )
 
+const ControllerEventName = "controller"
+
 type RepositoryConfig struct {
-	Resolutions []util.ResolutionConfiguration
 }
 
 // MetricRepository is an MetricUpdater which applies calls to update to all resolutions being tracked. It holds the
 // MetricStore instances for each resolution.
 type MetricRepository struct {
 	lock             sync.Mutex
 	resolutionStores map[string]*resolutionStores
+	exporter         exporter.EventExporter[UpdateSet]
 }
 
-func NewMetricRepository(config RepositoryConfig, factory MetricStoreFactory) *MetricRepository {
+func NewMetricRepository(
+	clusterID string,
+	resolutions []util.ResolutionConfiguration,
+	store storage.Storage,
+	storeFactory MetricStoreFactory,
+) *MetricRepository {
 	resoluationCollectors := make(map[string]*resolutionStores)
-
-	for _, resConf := range config.Resolutions {
-		resCollector, err := newResolutionStores(resConf, factory)
+	for _, resconf := range resolutions {
+		resolution, err := util.NewResolution(resconf)
+		if err != nil {
+			log.Errorf("failed to create resolution %s", err.Error())
+		}
+		resCollector, err := newResolutionStores(resolution, storeFactory)
 		if err != nil {
 			log.Errorf("NewMetricRepository: failed to init resolution metric: %s", err.Error())
 			continue
 		}
-		resoluationCollectors[resConf.Interval] = resCollector
+		resoluationCollectors[resolution.Interval()] = resCollector
 	}
 
 	repo := &MetricRepository{
 		resolutionStores: resoluationCollectors,
 	}
 
+	if store != nil {
+		pathFormatter, err := pathing.NewEventStoragePathFormatter("", clusterID, ControllerEventName)
+		if err != nil {
+			log.Errorf("filed to create path formatter for scrape controller: %s", err.Error())
+			return repo
+		}
+		encoder := exporter.NewJSONEncoder[UpdateSet]()
+		repo.exporter = exporter.NewEventStorageExporter(
+			pathFormatter,
+			encoder,
+			store,
+		)
+		// attempt to restore state from files
+		// get path of saved files
+		dirPath := path.Dir(pathFormatter.ToFullPath("", time.Time{}, ""))
```

Comment:
Feel free to add a `Dir()` to the PathFormatter interface. so you don't have to do this awkward empty call.

Reply:
Haha, yeah